modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4j1XmlLayout.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.log4j.internal;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.Node;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderFactory;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.core.layout.ByteBufferDestination;
+import org.apache.logging.log4j.core.layout.Encoder;
+import org.apache.logging.log4j.core.util.Transform;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.util.ReadOnlyStringMap;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * @author Hai Yu
+ * @see org.apache.log4j.layout.Log4j1XmlLayout
+ */
+@Plugin(
+	category = Node.CATEGORY, elementType = Layout.ELEMENT_TYPE,
+	name = Log4j1XmlLayout.PLUGIN_NAME, printObject = true
+)
+public final class Log4j1XmlLayout extends AbstractStringLayout {
```

Comment:
@yuhai Then let's just call it `XmlLayout`, no need to emphasize it's from Log4J 1.